web_src/js/features/find.js

```diff
@@ -0,0 +1,154 @@
+import {svg} from '../svg.js';
+const {appSubUrl, csrf} = window.config;
+
+const threshold = 50;
+let files = [];
+
+function hitAllKeys(keys, entry) {
+  let i = 0;
+  let j = 0;
+  const hitIndexes = [];
+  while (i < keys.length) {
+    for (; j < entry.length; j++) {
+      if (keys[i] === entry[j]) {
+        hitIndexes.push(j);
+        j++;
+        break;
+      }
+    }
+    i++;
+  }
+
+  return hitIndexes;
+}
+
+function addHighLightToHit(entry, indexes) {
+  let highLightText = '';
+  for (let i = 0; i < entry.length; i++) {
+    if (indexes.includes(i)) {
+      highLightText += `<b>${entry[i]}</b>`;
+    } else {
+      highLightText += entry[i];
+    }
+  }
+  return highLightText;
+}
+
+// function removeHighLight($a, entry) {
+//   $a.text(entry.replace(/(<([^>]+)>)/ig, ''));
+// }
+
+function filterRepoFiles(keys) {
+  if (keys.length > 0) {
+    // Remove all tr
+    $('#repo-find-files-table tbody').empty();
+
+    let hit = false;
+    const treeLink = $('#tree-link').val();
+    for (let i = 0; i < files.length; i++) {
+      if (i >= threshold) break;
```

Comment:
Does it `break` too early?

If there are 200 files loaded, then the first 50 (threshold) are not hit, will it break and result in an empty list?

Reply:
> Does it `break` too early?
> 
> If there are 200 files loaded, then the first 50 (threshold) are not hit, will it break and result in an empty list?

Sure,  I will fix it.